stacks-signer/src/signer.rs

```diff
@@ -1232,12 +1244,19 @@ impl Signer {
     /// DKG and participate in signing rounds accross crashes
     ///
     /// # Panics
-    /// Panics if the insertion fails
-    fn save_signer_state(&self) {
+    /// Panics if the encryption fails or if the insertion into stackerDB fails
```

Comment:
Neither of these should be panicking failures

Reply:
Also, I see that you're prepending a topic string to the cleartext.  Perhaps that topic string should be exposed in the `encrypt()` and `decrypt()` interfaces?  Then, `decrypt()` can detect bad decryption if the topic string is absent from the decrypted cleartext and return an appropriate "bad key" error.

Reply:
Nice point, I'll do that. I'm also happy to return a result rather than panicking. I'll improve the error handling here.

Reply:
Updated to return a result instead of panicking. I previously misunderstood the topic string suggestion. I'm only prefixing the private key when hashing to derive the encryption key. The message is encrypted as-is without any topic string. I can add one if you want, but that would make the persisted state in StackerDB larger.